### PR TITLE
Come up with STRNCPY_TERMINATED (#25).

### PR DESCRIPTION
**Pull request by marxin**
And use it in order to deal with new GCC 9 warnings.

### CODE REVIEW

**Comment by praeluceo** [comment]
Thank you for the PR, we are validating it now.

File: dcb_protocol.c [hunk 1]
```diff
@@ -2274,9 +2274,7 @@ cmd_status get_bwg_descrpt(char *device_name, u8 bwgid, char **name)
 				sizeof(char);
 			*name = (char*)malloc(size);
 			if (*name != NULL) {
-				strncpy(*name,
-					attribs.descript.pgid_desc[bwgid],
-					size); /* Localization OK */
+				memcpy(*name, attribs.descript.pgid_desc[bwgid], size); /* Localization OK */
```
Comment (dmertman) [inline dcb_protocol.c]: Why is this instance changed to a memcpy when all of the other ones were changed to use the new MACRO?

Comment (marxin) [reply dcb_protocol.c]: Doing that I would get:
```
In function ‘strncpy’,
    inlined from ‘get_bwg_descrpt’ at dcb_protocol.c:2277:5:
/usr/include/bits/string_fortified.h:106:10: error: ‘__builtin_strncpy’ output truncated before terminating nul copying as many bytes from a string as its length [-Werror=stringop-truncation]
  106 |   return __builtin___strncpy_chk (__dest, __src, __len, __bos (__dest));
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
dcb_protocol.c: In function ‘get_bwg_descrpt’:
dcb_protocol.c:2272:16: note: length computed here
 2272 |    size = (int)strlen(
      |                ^~~~~~~
 2273 |     attribs.descript.pgid_desc[bwgid]) +
      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
```

**Comment by marxin** [comment]
Just a note that GCC `9.1.0` has been just released.

**Comment by praeluceo** [comment]
This patch cleared internal validation.

**Pull request merged**